src/main/java/org/fenixedu/academic/ui/spring/service/AccountingReportService.java

```diff
@@ -0,0 +1,127 @@
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.fenixedu.academic.domain.accounting.AccountingTransactionDetail;
+import org.fenixedu.academic.domain.accounting.Event;
+import org.fenixedu.academic.domain.accounting.PaymentMethod;
+import org.fenixedu.academic.util.Bundle;
+import org.fenixedu.academic.util.Money;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.commons.spreadsheet.Spreadsheet;
+import org.joda.time.DateTime;
+import org.joda.time.Interval;
+import org.joda.time.LocalDate;
+import org.springframework.stereotype.Service;
+
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+
+/**
+ * Created by Sérgio Silva (hello@fenixedu.org).
+ */
+
+@Service
+public class AccountingReportService {
+
+    private final String DATE_PATTERN = "dd/MM/yyyy";
+    private final String DATE_TIME_PATTERN = "dd/MM/yyyy HH:mm:ss";
+
+    private boolean isFor(AccountingTransactionDetail atd, DateTime start, DateTime end) {
+        final DateTime whenRegistered = atd.getWhenRegistered();
+        return new Interval(start, end.plusDays(1)).contains(whenRegistered);
+    }
+
+    private void fill(Spreadsheet dateSheet, List<AccountingTransactionDetail> value) {
+        value.stream().sorted(Comparator.comparing(AccountingTransactionDetail::getWhenRegistered)).forEach(atd -> {
+            Event event = atd.getEvent();
+            Spreadsheet.Row row = dateSheet.addRow();
+
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.whenProcessed"), atd.getWhenProcessed().toString(DATE_TIME_PATTERN));
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.whenRegistered"), atd.getWhenRegistered().toString(DATE_TIME_PATTERN));
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.client"), event.getParty().getPartyPresentationName());
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.event.id"), event.getExternalId());
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.event.description"), event.getDescription().toString());
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.event.type"), BundleUtil.getString(Bundle.ENUMERATION, event.getEventType().getQualifiedName()));
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.payment.method"), atd.getPaymentMethod().getLocalizedName());
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.payment.reference"), atd.getPaymentReference());
+            row.setCell(BundleUtil.getString(Bundle.ACCOUNTING,"label.transaction.report.value"), atd.getTransaction().getAmountWithAdjustment().toPlainString());
+        });
+    }
+    
+    public Spreadsheet report(LocalDate start, LocalDate end) {
```

Comment:
Please refactor intermediate steps to improve readability of the the "main" report method.

OCD request: put private "auxiliar" methods at the bottom and the "main" method at the top of class